lib/src/git_backend.rs

```diff
@@ -493,39 +470,53 @@ impl GitBackend {
             }
         }
         let parent_refs = parents.iter().collect_vec();
-        let git_id = locked_repo
-            .commit(
-                Some(&create_no_gc_ref()),
-                &author,
-                &committer,
-                message,
-                &git_tree,
-                &parent_refs,
-            )
-            .map_err(|err| BackendError::WriteObject {
-                object_type: "commit",
-                source: Box::new(err),
-            })?;
-        let id = CommitId::from_bytes(git_id.as_bytes());
-        let extras = serialize_extras(contents);
-        let mut mut_table = self
-            .extra_metadata_store
-            .get_head()
-            .unwrap()
-            .start_mutation();
-        if let Some(existing_extras) = mut_table.get_value(git_id.as_bytes()) {
-            if existing_extras != extras {
-                return Ok(None);
+        loop {
+            let git_id = locked_repo
+                .commit(
+                    Some(&create_no_gc_ref()),
+                    &author,
+                    &committer,
+                    message,
+                    &git_tree,
+                    &parent_refs,
+                )
+                .map_err(|err| BackendError::WriteObject {
+                    object_type: "commit",
+                    source: Box::new(err),
+                })?;
+            let id = CommitId::from_bytes(git_id.as_bytes());
+            let extras = serialize_extras(contents);
+            let mut mut_table = self
+                .extra_metadata_store
+                .get_head()
+                .unwrap()
+                .start_mutation();
+            if let Some(existing_extras) = mut_table.get_value(git_id.as_bytes()) {
```

Comment:
Minor: If it's not too difficult, I think it would be nice to make the loop shorter.

I think this should work:

```rust
  match mut_table.get_value(git_id.as_bytes()) {
    Some(existing_extras) if existing_extras != extras { /* fudge timestamp */ }
    _ => break,
  }
} // end `loop`

mut_table.add_entry( /*...*/);
// Rest of function, up to `Ok(id)`
```

Reply:
Yes, I agree. It requires using a `break id` to give the whole `loop` expression a value (or I could have made `id` a mutable variable).  I've made that change, and I also moved out `extras` and `mut_table` outside the loop since I realized they're also not changing.

Reply:
Looks great to me!